examples/spineml_braitenberg/spineml_braitenberg.cc

```diff
@@ -0,0 +1,204 @@
+// BoB robotics includes
+#include "common/circstat.h"
+#include "common/collision.h"
+#include "common/main.h"
+#include "common/read_objects.h"
+#include "common/sfml_world.h"
+#include "hid/joystick.h"
+#include "robots/simulated_tank.h"
+
+// SpineML simulator includes
+#include "simulator.h"
+
+// Third-party includes
+#include "third_party/units.h"
+
+// Standard C++ includes
+#include <chrono>
+#include <iostream>
+#include <thread>
+
+using namespace BoBRobotics;
+using namespace std::literals;
+using namespace units::angle;
+using namespace units::length;
+using namespace units::math;
+using namespace units::literals;
+
+
+class ArenaObject : public sf::Drawable
+{
+public:
+    template<class VectorArrayType, class MatrixType>
+    ArenaObject(const SFMLWorld<> &renderer, const VectorArrayType &original, const MatrixType &resized)
+      : m_GreenShape(original.size()) , m_RedShape(original.size())
+    {
+        // Dark green
+        m_GreenShape.setFillColor(sf::Color{ 0x00, 0x88, 0x00 });
+
+        // Add each vertex to the shape
+        for (size_t i = 0; i < original.size(); i++) {
+            m_GreenShape.setPoint(i, renderer.vectorToPixel(original[i]));
+        }
+
+        m_RedShape.setFillColor(sf::Color::Red);
+        for (size_t i = 0; i < original.size(); i++) {
+            m_RedShape.setPoint(i, renderer.vectorToPixel(resized(i, 0), resized(i, 1)));
+        }
+    }
+
+    virtual void draw(sf::RenderTarget &target, sf::RenderStates states) const override
+    {
+        target.draw(m_RedShape, states);
+        target.draw(m_GreenShape, states);
+    }
+
+private:
+    sf::ConvexShape m_GreenShape, m_RedShape;
+};
+
+
+int bob_main(int, char **)
+{
+    // The x and y dimensions of the robot
+    // **THINK** if would be nice if this was part of a robot class?
```

Comment:
One could certainly add this to the robot class, but the only thing is that a robot of a given class is not always going to be one size (e.g. if we mod the Lego robot, or with the ``TankNetSink`` class, which can be anything)

Reply:
Wouldn't the same argument apply to axle length?

Reply:
Heh, good point. We could replace the current ``getRobotWidth()`` function with a ``getRobotSize()`` (it could return e.g. a ``Vector3``)

Reply:
Shall we leave this for a subsequent PR?

Reply:
Yeah, let's also put off this one till later.

Reply:
#120